### PR TITLE
feat: fix filters attribute name

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

As seen in https://github.com/strapi/strapi/issues/21338 , attribute names called `filters` broke the populate validation.
That's cause it was confusing a "filter" inside a populate with the filters attribute name:
```ts
{
  populate: {
    foo: {
      filters: { bar: { $eq: 1 } }  // Should filter entries where bar is 1
    }
  }
}

{
  populate: {
    foo: {
      populate: {
        filters: true  //  Should populate the "filters" attribute
      }
    }
  }
}
```

The `traverseQueryPopulate` was assuming `filters` was an attribute, as it was an actual attribute of the schema. Hence the first query was always invalid.

This PR solves this by modifying the `traverseQueryPopulate` and tracking each traversed node parent.

### CODE REVIEW

**Comment by Marc-Roig** [comment]
@alexandrebodin @innerdvations  @Convly I think this might be easier if we spend 15 minutes on discord discussing this ticket together 🤞

**Comment by Convly** [comment]
Agree, lots of things to say :joy:

**Comment by Marc-Roig** [comment]
@Convly @alexandrebodin if we agree this is what we want to merge, could we have a last look and proceed with it? 🙏

File: packages/core/utils/src/traverse/query-populate.ts [hunk 1]
```diff
@@ -24,53 +24,63 @@ const isKeyword = (keyword: string) => {
...
   isArray(value) && value.every(isString);
 
 const isObj = (value: unknown): value is Record<string, unknown> => isObject(value);
 
 const populate = traverseFactory()
-  // Array of strings ['foo', 'foo.bar'] => map(recurse), then filter out empty items
+  .intercept(isPopulateString, async (visitor, options, populate, { recurse }) => {
+    const populateObject = pathsToObjectPopulate([populate]);
+    const traversedPopulate = (await recurse(visitor, options, populateObject)) as PopulateObject;
+    const [result] = objectPopulateToPaths(traversedPopulate);
```
Comment (Convly) [inline packages/core/utils/src/traverse/query-populate.ts:42]: Could we add some context to this bit to make sure it's not altered by future contributors? Basically explaining why we have to transform in & out the strings

File: packages/core/utils/src/__tests__/query-populate.test.ts [hunk 2]
```diff
@@ -104,4 +104,75 @@ describe('traverseQueryPopulate', () => {
...
+          break;
+
+        case 'filters':
+          // Parent information should be available
+          expect(parent.key).toBe('address');
+          expect(parent.attribute).not.toBeUndefined();
+          break;
+        default:
+          break;
+      }
```
Comment (Convly) [inline packages/core/utils/src/__tests__/query-populate.test.ts:169]: Could we also add an expect on the attribute itself? To make sure both have the correct values?

```suggestion
    await traverseQueryPopulate(({ key, parent, attribute }) => {
      switch (key) {
        case 'address':
          // top level populate should not have parent
          expect(parent).toBeUndefined();
          expect(attribute).toBeDefined();
          break;

        case 'filters':
          // Parent information should be available
          expect(parent.key).toBe('address');
          expect(parent.attribute).not.toBeUndefined();
          expect(attribute).toBeDefined();
          break;
        default:
          break;
      }
```

**Review by Convly — COMMENTED** [review]
Seems like unit tests are failing, the rest lgtm

**Review by Convly — APPROVED** [review]

**Pull request merged**